core/dbt/cli/main.py

```diff
@@ -453,6 +469,30 @@ def test(ctx, **kwargs):
     return None, True
 
 
+def _get_config_from_ctx(ctx):
+    return RuntimeConfig.from_parts(
+        ctx.obj["project"],
+        ctx.obj["profile"],
+        ctx.obj["flags"],
+    )
+
+
+def _set_manifest_on_task(task, ctx, write_perf_info=False):
```

Comment:
Why do we choose this over passing in manifest as an object during initialization of task

Reply:
I initial thought there wasn't much of a difference in either approach, but after speaking with @MichelleArk I believe setting the manifest directly on task init is favorable given users may attempt to set a different manifest after a run, which may have unknown consequences given how stateful our task classes are.

Reply:
Makes sense! Yes I agree seems like re-create the whole task would be a cleaner interface than setting things after it being constructed!